### PR TITLE
ClinSeq::LOH - Write results to temp dir first.

### PR DESCRIPTION
**Pull request by gatoravi**
This PR modifies this module to write the results
to a temp dir and copy the results over at the end for
better performance.

### CODE REVIEW

**Comment by brummett** [comment]
As far as I can tell, all the I/O is writes to the output directory, whether it's to the final ```outdir``` or the temp directory.  Are there reads from that directory I'm missing?

If there's no reads, and it's writing the same amount of data either way, how is it faster?

**Comment by gatoravi** [comment]
Sorry, that was a stray variable. `$temp_outdir` is the one that is actually used. I've removed it now. Writes now happen on `tmp` and results copied over at the end.

**Comment by gatoravi** [comment]
Wouldn't writing to local temp space on a blade and then copying over to disk be faster if we were constrained by disk bandwidth?

**Comment by brummett** [comment]
The same disk contention is going to still exist whether it's in the old code with the tools writing directly to the output directory, or in the new code copying to the output directory.

One case where it's better to write to /tmp locally first is if it writes some data to a file and reads it back in during the same step.  Then, the reads are taking place locally on the same machine rather than going over the network.

**Comment by malachig** [comment]
Maybe it was my imagination but it seemed like there were files there during processing that did not exist in the final result.  If these files are not part of the final result then wouldn't there be an advantage to writing to temp and them moving the final result over (that way the intermediate files are never going over the network)?

File: lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm [hunk 1]
```diff
@@ -255,11 +262,12 @@ sub filter_loh {
 #Get rid of intermediate files
 sub cleanup {
     my $self = shift;
+    my $outdir = shift;
     my @patterns = qw(*.Somatic* *.readcounts *.hc *.LOH
```
Comment (malachig) [inline lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm]: Do we actually need all of these files in the final result under non debugging conditions?

**Comment by gatoravi** [comment]
Sorry, I see your point now @brummett .
One point is we do a cleanup on the temp dir before the copy so some chunky readcount files are removed before the copy.

File: lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm (hunk 1 above)
Comment (gatoravi) [inline lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm]: I don't think so and hence I got rid of them. What do you think?

**Comment by brummett** [comment]
I get it now.  If it's not copying all the files - that makes sense.

+1

File: lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm (hunk 1 above)
Comment (malachig) [inline lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm]: Right I see now that these are the files being removed and not kept.  Duh.  This makes sense!

**Comment by malachig** [comment]
+1

**Pull request merged**